api/proto/teleport/legacy/client/proto/event.proto

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/api/client/proto";
+
+// Operation identifies type of operation
+enum Operation {
+  // INIT is sent as a first sentinel event
+  // on the watch channel
+  INIT = 0;
+  // PUT identifies created or updated object
+  PUT = 1;
+  // DELETE identifies deleted object
+  DELETE = 2;
+}
+
+// Event returns cluster event
+message Event {
+  reserved 7;
+
+  // Operation identifies operation
+  Operation Type = 1;
+  // Resource contains the updated resource
+  oneof Resource {
```

Comment:
This now a mix of Google and Gogo protos, correct?

On one hand I'm happy to see less Gogo, and this seems to work. OTOH, mixing the libs like makes me a bit uneasy - for example, is Gogo guaranteed to work with "googleproto.Equal" and other funcs?

Reply:
That's correct. It looks like `googleproto.Equal` won't work with raw gogo messages because they're V1 messages and not V2. You could use `protoadapt.MessageV2Of`, however.

Reply:
Losing the ability to cleanly `proto.Equal` (and potentially other funcs) seems pretty bad to me.

Reply:
Hrm. I think I'm wrong about this actually -- let me poke at this a bit more. I may have just been mis-using my test messages.

Reply:
Okay, so `proto.Equal` still works with the new output object. So it looks like this indeed keeps the existing functionality.

Reply:
Do you mind adding a test that exercises Google's proto.Equal against a "mixed" proto? I know it sounds a bit strange, but I want something to break cleanly if a future dependency causes problems, for example.

Reply:
I've added in this test for the event object. I'll expand it in https://github.com/gravitational/teleport/pull/29119, which will make this event a "mixed" object to prove it out.